ide/refactoring.api/src/org/netbeans/modules/refactoring/spi/impl/RefactoringPanel.java

```diff
@@ -784,6 +781,25 @@ public void run() {
                                                     root.setNodeLabel(description + getErrorDesc(0, occurrences, hiddenOccurrences, isQuery && sizeIsApproximate.get()));
                                                     if (last) {
                                                         tree.repaint();
+                                                        if (showParametersPanel) {
+                                                            if (size.get() < MAX_ROWS) {
+                                                                expandAll();
+                                                                if (isQuery) {
+                                                                    selectNextUsage();
+                                                                }
+                                                            } else {
+                                                                expandButton.setSelected(false);
+                                                            }
+                                                        } else {
+                                                            if (expandButton.isSelected()) {
+                                                                expandAll();
+                                                                if (isQuery) {
+                                                                    selectNextUsage();
+                                                                }
+                                                            } else {
+                                                                expandButton.setSelected(false);
+                                                            }
+                                                        }
```

Comment:
I am not sure what this does. When is this called?

Reply:
This is where the first element in the tree is clicked to show a preview. This code is called after the tree is filled with elements.

Reply:
Here is a similar code for refactoring. 
https://github.com/apache/netbeans/blob/ec0ab034a624aa4efac1e2add97fae591f04102a/ide/refactoring.api/src/org/netbeans/modules/refactoring/spi/impl/RefactoringPanel.java#L911-L946

The only difference is that for refactoring the tree is already filled before it is displayed.

Reply:
ok. I believe what makes this method so confusing to me is that it has the
```
if (isQuery && showParametersPanel) {
```
section in the middle, and then again a 
```
if (!(isQuery && showParametersPanel)) {
```
in the end. And the last if does call `setupTree()` which has similar code to this section.

We could move a few things around to simplify it:
```diff
diff --git a/ide/refactoring.api/src/org/netbeans/modules/refactoring/spi/impl/RefactoringPanel.java b/ide/refactoring.api/src/org/netbeans/modules/refactoring/spi/impl/RefactoringPanel.java
index 7e2bff3..e1f02c6 100644
--- a/ide/refactoring.api/src/org/netbeans/modules/refactoring/spi/impl/RefactoringPanel.java
+++ b/ide/refactoring.api/src/org/netbeans/modules/refactoring/spi/impl/RefactoringPanel.java
@@ -781,25 +781,6 @@
                                                     root.setNodeLabel(description + getErrorDesc(0, occurrences, hiddenOccurrences, isQuery && sizeIsApproximate.get()));
                                                     if (last) {
                                                         tree.repaint();
-                                                        if (showParametersPanel) {
-                                                            if (size.get() < MAX_ROWS) {
-                                                                expandAll();
-                                                                if (isQuery) {
-                                                                    selectNextUsage();
-                                                                }
-                                                            } else {
-                                                                expandButton.setSelected(false);
-                                                            }
-                                                        } else {
-                                                            if (expandButton.isSelected()) {
-                                                                expandAll();
-                                                                if (isQuery) {
-                                                                    selectNextUsage();
-                                                                }
-                                                            } else {
-                                                                expandButton.setSelected(false);
-                                                            }
-                                                        }
                                                     }
                                                 }
                                             }
@@ -845,10 +826,11 @@
                     }
                     
                     if (!(isQuery && showParametersPanel)) {
-                        root.setNodeLabel(description + getErrorDesc(errorsNum, isQuery ? size.get() : elements.size(), hidden, isQuery && sizeIsApproximate.get()).toString());
+                        root.setNodeLabel(description + getErrorDesc(errorsNum, elements.size(), hidden, false).toString());
                         setupTree(root, showParametersPanel, elements.size());
+                    } else if (isQuery && showParametersPanel) {
+                        SwingUtilities.invokeLater(() -> expandTreeIfNeeded(showParametersPanel, size.get()));
                     }
-                    
                 }
 
                 private StringBuffer getErrorDesc(int errorsNum, int occurencesNum, int hiddenNum, boolean occurencesNumApproximate) throws MissingResourceException {
@@ -916,25 +898,7 @@
                 createTree(root);
 
                 splitPane.setDividerLocation(0.3);
-                if (showParametersPanel) {
-                    if (size < MAX_ROWS) {
-                        expandAll();
-                        if (!isQuery) {
-                            selectNextUsage();
-                        }
-                    } else {
-                        expandButton.setSelected(false);
-                    }
-                } else {
-                    if (expandButton.isSelected()) {
-                        expandAll();
-                        if (!isQuery) {
-                            selectNextUsage();
-                        }
-                    } else {
-                        expandButton.setSelected(false);
-                    }
-                }
+                expandTreeIfNeeded(showParametersPanel, size);
 
                 tree.setSelectionRow(0);
                 setRefactoringEnabled(true, true);
@@ -944,7 +908,25 @@
             }
         });
     }
-    
+
+    private void expandTreeIfNeeded(boolean showParametersPanel, int size) {
+        if (showParametersPanel) {
+            if (size < MAX_ROWS) {
+                expandAll();
+                selectNextUsage();
+            } else {
+                expandButton.setSelected(false);
+            }
+        } else {
+            if (expandButton.isSelected()) {
+                expandAll();
+                selectNextUsage();
+            } else {
+                expandButton.setSelected(false);
+            }
+        }
+    }
+
      private Map<FileObject, Long> timeStamps = new HashMap<FileObject, Long>();
      
      private void storeTimeStamps(Set<FileObject> fileObjects) {

```

Reply:
Made the change, thanks!